### PR TITLE
add Kolors model 

### PR DESCRIPTION
**Pull request by nc-BobLee**

### CODE REVIEW

File: examples/language-modeling/README.md [hunk 1]
```diff
@@ -391,7 +391,7 @@ PT_TE_CUSTOM_OP=1 PT_HPU_LAZY_MODE=0 python ../gaudi_spawn.py \
     --torch_compile_backend hpu_backend \
     --torch_compile \
     --fp8 \
-    --use_flash_attention True \
+    --attn_implementation gaudi_fused_sdpa \
     --flash_attention_causal_mask True  \
```
Comment (Wei-Lin-Intel) [inline examples/language-modeling/README.md:395]: Do not change these unrelated parts.

File: examples/language-modeling/README.md [hunk 2]
```diff
@@ -435,7 +435,7 @@ python3 ../gaudi_spawn.py --use_deepspeed --world_size 8 run_lora_clm.py \
   --lora_rank 4 \
   --lora_target_modules "q_proj" "v_proj" "k_proj" "o_proj" \
   --validation_split_percentage 4 \
-  --use_flash_attention True \
+  --attn_implementation gaudi_fused_sdpa \
```
Comment (Wei-Lin-Intel) [inline examples/language-modeling/README.md:438]: Do not change these unrelated parts.

File: examples/language-modeling/README.md [hunk 3]
```diff
@@ -478,7 +478,7 @@ python3 ../gaudi_spawn.py --world_size 8 --use_mpi run_lora_clm.py \
   --torch_compile_backend hpu_backend \
   --torch_compile \
   --gradient_accumulation_steps 2 \
-  --use_flash_attention True \
+  --attn_implementation gaudi_fused_sdpa \
   --flash_attention_causal_mask True
```
Comment (Wei-Lin-Intel) [inline examples/language-modeling/README.md:482]: Do not change these unrelated parts.

File: examples/language-modeling/README.md [hunk 4]
```diff
@@ -530,7 +530,7 @@ The format of the text files (with extensions .text or .txt) is expected to be
 
 To run prompt tuning finetuning, you can use `run_prompt_tuning_clm.py`.
 Here are single-card command examples for Llama2-7B:
-- single-card finetuning of meta-llama/Llama-2-7b-hf with dataset "ought/raft" and config "twitter_complaints":
+- single-card finetuning of meta-llama/Llama-2-7b-hf with dataset "regisss/raft" and config "default":
 ```bash
```
Comment (Wei-Lin-Intel) [inline examples/language-modeling/README.md:534]: Do not change these unrelated parts.

File: examples/language-modeling/README.md [hunk 5]
```diff
@@ -647,7 +647,7 @@ PT_HPU_LAZY_MODE=1 python3 ../gaudi_spawn.py  \
         --eval_strategy epoch \
         --pipelining_fwd_bwd \
         --use_lazy_mode \
-        --use_flash_attention True \
+        --attn_implementation gaudi_fused_sdpa \
         --deepspeed llama3_ds_zero1_config.json \
```
Comment (Wei-Lin-Intel) [inline examples/language-modeling/README.md:651]: Do not change these unrelated parts.

File: examples/language-modeling/run_clm.py [hunk 6]
```diff
@@ -766,7 +766,7 @@ def compute_metrics(eval_preds):
         metrics = trainer.evaluate()
 
         if data_args.streaming:
-            metrics["eval_samples"] = max_eval_samples
+            metrics["eval_samples"] = training_args.max_steps * training_args.per_device_eval_batch_size
         else:
```
Comment (Wei-Lin-Intel) [inline examples/language-modeling/run_clm.py:770]: Do not change these unrelated parts.

File: examples/stable-diffusion/run_kolors_demo.sh [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export PT_HPU_LAZY_MODE=1
+export PT_HPU_GPU_MIGRATION=1
+
+python3 ./text_to_image_kolors.py \
+    --model_name_or_path '/mnt/ceph1/libo/hf_models/Kolors' \
+    --prompts "一张瓢虫的照片，微距，变焦，高质量，电影，拿着一个牌子，写着“可图”" \
```
Comment (Wei-Lin-Intel) [inline examples/stable-diffusion/run_kolors_demo.sh:8]: Do not use such model path

File: examples/stable-diffusion/text_to_image_kolors.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+    pipe = GaudiStableDiffusionKolorsPipeline(
+            vae=vae,
+            text_encoder=text_encoder,
+            tokenizer=tokenizer,
+            unet=unet,
+            scheduler=scheduler,
+            **kwargs,)
+    pipe = pipe.to("cuda")
+
```
Comment (Wei-Lin-Intel) [inline examples/stable-diffusion/text_to_image_kolors.py:59]: cuda?

File: examples/stable-diffusion/text_to_image_kolors.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+            prompt=args.prompts,
+            height=1024,
+            width=1024,
+            num_inference_steps=3,
+            guidance_scale=5.0,
+            num_images_per_prompt=1,
+            generator= torch.Generator(pipe.device).manual_seed(878))
+    torch.hpu.synchronize()
+    #exit()
+
```
Comment (Wei-Lin-Intel) [inline examples/stable-diffusion/text_to_image_kolors.py:72]: Remove `#exit()`

File: examples/stable-diffusion/text_to_image_kolors.py [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        num_inference_steps=50,
+        guidance_scale=5.0,
+        num_images_per_prompt=1,
+        is_profiler = False,
+        generator= torch.Generator(pipe.device).manual_seed(5554)).images[0]
+
+    image.save(f'piaocong.jpg')
+
+
+
```
Comment (Wei-Lin-Intel) [inline examples/stable-diffusion/text_to_image_kolors.py:86]: Too many spaces here

File: examples/text-generation/docker/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,34 @@
+# Parameterize base image components
```
Comment (Wei-Lin-Intel) [inline examples/text-generation/docker/Dockerfile:1]: Where does this file come?

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py [hunk 12]
```diff
@@ -0,0 +1,631 @@
...
+
+
+def setup_profile(steps):
+    activities = [torch.profiler.ProfilerActivity.CPU, torch.profiler.ProfilerActivity.HPU]
+    profiler=torch.profiler.profile(
+       schedule=torch.profiler.schedule(wait=0, warmup=1, active=steps, repeat=1),
+       activities=activities,
+       with_stack=True,
+        on_trace_ready=torch.profiler.tensorboard_trace_handler("/mnt/ceph1/libo/kolors/profile/", use_gzip=True))
+    return profiler
```
Comment (Wei-Lin-Intel) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py:35]: Do not use such model path

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py [hunk 13]
```diff
@@ -0,0 +1,631 @@
...
+    rope_cache = rope_cache[:sq]
+    xshaped = x.reshape(sq, -1, np, rot_dim // 2, 2)
+    rope_cache = rope_cache.view(sq, -1, 1, xshaped.size(3), 2)
+    x_out2 = torch.stack(
+        [
+            xshaped[..., 0] * rope_cache[..., 0] - xshaped[..., 1] * rope_cache[..., 1],
+            xshaped[..., 1] * rope_cache[..., 0] + xshaped[..., 0] * rope_cache[..., 1],
+        ],
+        -1,
+    )
```
Comment (Wei-Lin-Intel) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py:132]: Why not using hpu fused rope?

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py [hunk 14]
```diff
@@ -0,0 +1,631 @@
...
+    return torch.cat((x_out2, x_pass), dim=-1)
+
+
+def forward_gaudi(self, query_layer, key_layer, value_layer, attention_mask):
+    pytorch_major_version = int(torch.__version__.split('.')[0])
+    if pytorch_major_version >= 2:
+        query_layer, key_layer, value_layer = [k.permute(1, 2, 0, 3) for k in [query_layer, key_layer, value_layer]]
+        if attention_mask is None and query_layer.shape[2] == key_layer.shape[2]:
+            context_layer = torch.nn.functional.scaled_dot_product_attention(query_layer, key_layer, value_layer,
+                                                                             is_causal=True)
```
Comment (Wei-Lin-Intel) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py:143]: If this path is valid, use `fused_spda` instead. If not, then no need to maintain such path.

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py [hunk 15]
```diff
@@ -0,0 +1,631 @@
...
+        else:
+            if attention_mask is not None:
+                attention_mask = ~attention_mask
+            #context_layer = torch.nn.functional.scaled_dot_product_attention(query_layer, key_layer, value_layer,
+            #                                                                 attention_mask)
+            fsdpa_mode="None"
+            from habana_frameworks.torch.hpex.kernels import FusedSDPA
```
Comment (Wei-Lin-Intel) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_kolors.py:150]: Why `none` not `fast`? Accuracy issue?

File: optimum/habana/transformers/training_args.py [hunk 16]
```diff
@@ -942,6 +957,14 @@ def __post_init__(self):
...
             )
             os.environ["FLASH_ATTENTION_FAST_SOFTMAX"] = "1"
+        if self.flash_attention_causal_mask:
+            assert self.attn_implementation == "gaudi_fused_sdpa", (
+                "flash_attention_causal_mask works only with --attn_implementation gaudi_fused_sdpa"
+            )
+        if self.flash_attention_fp8:
+            assert self.attn_implementation == "gaudi_fused_sdpa", (
+                "flash_attention_causal_mask works only with --attn_implementation gaudi_fused_sdpa"
+            )
```
Comment (Wei-Lin-Intel) [inline optimum/habana/transformers/training_args.py:968]: Please double check why this PR would change a lot of files for training. We should use oh-fork:aice/v1.22.0 as the base repo.

**Review by Wei-Lin-Intel — CHANGES_REQUESTED** [review]

**Comment by Wei-Lin-Intel** [comment]
Please run `make style` to reformat
